frontend/server/src/Controllers/Run.php

```diff
@@ -915,6 +921,144 @@ public static function apiDetails(\OmegaUp\Request $r): array {
             $response['judged_by'] = strval($run->judged_by);
         }
 
+        $problemArtifacts = new \OmegaUp\ProblemArtifacts(
+            $problem->alias,
+            $run->commit
+        );
+
+        $cases = self::getCases(
+            $problemArtifacts,
+            'cases',
+            $problem->alias,
+            $run->commit
+        );
+        /** @var int $responseSize */
+        $responseSize = array_reduce(
+            $cases,
+            /**
+             * @param array{mode: int, type: string, id: string, size: int, path: string} $case
+             */
+            function (?int $sum, $case): int {
+                if (is_null($sum)) {
+                    return $case['size'];
+                }
+                $sum += $case['size'];
+                return $sum;
+            }
+        );
+
+        if (
+            $problem->show_diff === \OmegaUp\ProblemParams::NO_SHOW_DIFFS
+            || $responseSize > 4096 // Forcing to hide diffs when inputs/outpus exceed 4kb
+        ) {
+            $response['show_diff'] = \OmegaUp\ProblemParams::NO_SHOW_DIFFS;
+            return $response;
+        }
+        if ($problem->show_diff === \OmegaUp\ProblemParams::SHOW_ALL_DIFFS) {
+            $dataCases = self::getProblemCases(
+                $problemArtifacts,
+                'cases',
+                $problem->alias,
+                $run->version
+            );
+            $response['cases'] = $dataCases;
```

Comment:
se le puede dar inline a esta variable? onda

```suggestion
            $response['cases'] = self::getProblemCasesContents(
```

Reply:
Listo